### PR TITLE
Add link to release notes (fixes #159)

### LINKED ISSUE #159: Release page does not include release notes
The release page should include important release notes.

e.g known issues, major features / bug fixes.

### PR DESCRIPTION
**Pull request by neiljerram**

### CODE REVIEW

**Comment by neiljerram** [comment]
@caseydavenport Here is an approach for fixing #159 - what do you think?

(Obviously we should add links to release tags for all components, not just Felix.)

**Comment by caseydavenport** [comment]
@neiljerram seems like a sensible approach to me.

Do you think there might be scope for a brief overview of new significant features listed on this page itself in addition to pointing users to go to each of the subcomponents?

**Comment by neiljerram** [comment]
@caseydavenport Not sure.  My guess/feeling is that the page _might_ be overly cluttered if we included _all_ the release notes for each component that.

However, I see (when adding these links) that _some_ of our components have a nice one line summary of what's in the release (in the release title), and it probably would be nice to include that on the releases page.   (But the Felix release doesn't yet have that summary.)

Anyway, I've added the release note links now for components other than Felix.  How about merging that as a useful increment?

File: v2.0/releases/index.md [hunk 1]
```diff
@@ -8,16 +8,19 @@ Use the version selector at the top-right of this page to view a different relea
...
-| felix                  | 2.0.0-beta.3 |
-| calicoctl              | v1.0.0-beta |
-| calico/node            | v1.0.0-beta |
-| calico/cni             | v1.5.0 |
-| libcalico              | v0.18.0 |
-| calico-bird            | v0.1.0  |
-| libnetwork-plugin      | v1.0.0-beta |
-| calico/kube-policy-controller | v0.5.0 |
+| Component                     | Version                                                                                    |
+|-------------------------------+--------------------------------------------------------------------------------------------|
```
Comment (caseydavenport) [inline v2.0/releases/index.md]: I think the table columns got messed up a bit.  You're missing a `|`.

I prefer to not make the separators stretch the full width of the table (makes it harder to read and isn't necessary for it to format properly).

You should be able to leave the column headers unchanged in this PR.

**Comment by caseydavenport** [comment]
Yep, SGTM.

Looks like you'll need to rebase + one more comment from me (looks like the table formatting is broken).

Sorry I took so long to get back to this one

**Comment by neiljerram** [comment]
Thanks.  Not sure exactly what you mean by the formatting (perhaps a '+' where you might expect '|'), but it's generated by a convenient Emacs mode, and still generates correct output, and is nice to read in source form too.  So I think best to keep like this.

**Comment by caseydavenport** [comment]
Yeah, I think I just read the table wrong.  Looks good!

**Pull request merged**